web/src/stores/workflow-store.ts

```diff
@@ -372,6 +379,19 @@ export const useWorkflowStore = create<WorkflowStoreState>((set, get) => ({
 			}),
 		);
 	},
+	addMissingSecretByIdx: (secret: string, nodeIdx: number) =>
+		set(
+			produce((draft) => {
+				draft.missingSecretForNodes[nodeIdx] = secret;
```

Comment:
Interesting. Just learned that for out-of-bounds access the array automatically grows and initializes the new fields with `undefined`. Since you are relying on this, adding new nodes will always display an error regarding the not available secrets (see my start node comment)